<?xml version="1.0"?>
<launch>
  <arg name="lin_kp" default="500.0"/>
  <arg name="lin_ki" default="0.0"/>
  <arg name="lin_kd" default="0.0"/>
  <arg name="yaw_kp" default="500.0"/>
  <arg name="yaw_ki" default="0.0"/>
  <arg name="yaw_kd" default="0.0"/>
  <arg name="max_linear_windup" default="100.0"/>
  <arg name="max_yaw_windup" default="100.0"/>
  <arg name="motor_command_limit" default="1000.0"/>
  <arg name="use_vel_fitting" default="false"/>
  <arg name="negative_scaling_factor" default="1.0"/>
  <arg name="positive_scaling_factor" default="1.0"/>
  <!-- Twist to Drive receives the twist messages, and outputs motor drive commands -->
  <node pkg="rowbot_control" type="course_to_drive" name="course_to_drive">
    <param name="~lin_kp" value="$(arg lin_kp)"/>
    <param name="~lin_ki" value="$(arg lin_ki)"/>
    <param name="~lin_kd" value="$(arg lin_kd)"/>
    <param name="~yaw_kp" value="$(arg yaw_kp)"/>
    <param name="~yaw_ki" value="$(arg yaw_ki)"/>
    <param name="~yaw_kd" value="$(arg yaw_kd)"/>
    <param name="~max_linear_windup" value="$(arg max_linear_windup)"/> <!-- The cap on the i-term of the PID controller -->
    <param name="~max_yaw_windup" value="$(arg max_yaw_windup)"/> <!-- The cap on the i-term of the PID controller -->
    <param name="~motor_command_limit" value="$(arg motor_command_limit)"/>
    <param name="~use_vel_fitting" value="$(arg use_vel_fitting)"/>
    <param name="~negative_scaling_factor" value="$(arg negative_scaling_factor)"/>
    <param name="~positive_scaling_factor" value="$(arg positive_scaling_factor)"/>
</node>
</launch>
